Sources/SourceKitLSP/Swift/SwiftTestingScanner.swift

```diff
@@ -321,36 +334,34 @@ fileprivate extension AttributeSyntax {
 }
 
 fileprivate extension MemberAccessExprSyntax {
-  /// The base name of this instance, i.e. the string value of `base` joined
-  /// with any preceding base names.
+  /// The fully-qualified name of this instance (subject to available
+  /// information.)
   ///
-  /// For example, if this instance represents the expression `x.y.z(123)`,
-  /// the value of this property is `"x.y"`. If the value of `base` is `nil`,
-  /// the value of this property is also `nil`.
-  var baseName: String? {
-    if let declReferenceExpr = base?.as(DeclReferenceExprSyntax.self) {
-      return declReferenceExpr.baseName.text
-    } else if let baseMemberAccessExpr = base?.as(MemberAccessExprSyntax.self) {
-      if let baseBaseName = baseMemberAccessExpr.baseName {
-        return "\(baseBaseName).\(baseMemberAccessExpr.declName.baseName.text)"
-      }
-      return baseMemberAccessExpr.declName.baseName.text
-    }
-
-    return nil
+  /// The value of this property are all the components of the based name
+  /// name joined together with `.`.
+  var fullyQualifiedName: String {
+    components.joined(separator: ".")
   }
 
-  /// The fully-qualified name of this instance (subject to available
+  /// The name components of this instance (subject to available
   /// information.)
   ///
-  /// The value of this property is this instance's `baseName` property joined
-  /// with its `name` property. For example, if this instance represents the
-  /// expression `x.y.z(123)`, the value of this property is `"x.y.z"`.
-  var fullyQualifiedName: String {
-    if let baseName {
-      return "\(baseName).\(declName.baseName.text)"
+  /// The value of this property is this base name of this instance,
+  /// i.e. the string value of `base` preceeded with any preceding base names
+  /// and followed by its `name` property.
+  ///
+  /// For example, if this instance represents
+  /// the expression `x.y.z(123)`, the value of this property is
+  /// `["x", "y", "z"]`.
+  var components: [String] {
+    if let declReferenceExpr = base?.as(DeclReferenceExprSyntax.self) {
+      return [declReferenceExpr.baseName.text, declName.baseName.text]
+    } else if let baseMemberAccessExpr = base?.as(MemberAccessExprSyntax.self) {
+      let baseBaseNames = baseMemberAccessExpr.components.dropLast()
+      return baseBaseNames + [baseMemberAccessExpr.declName.baseName.text, declName.baseName.text]
```

Comment:
Why do you need to `dropLast` and then add the `baesName.text` again? Shouldn’t the following be sufficient? Or am I missing something obvious?

```suggestion
      return baseMemberAccessExpr.components + [declName.baseName.text]
```

Reply:
Good catch, I've updated.